### ISSUE TITLE
energy returned to grid error in Energy Day Graph

### ISSUE THREAD

**Issue by Ron1926** [magico13/ha-emporia-vue#211]
The Emporia Vue Home Assistant (HA) integration is a fantastic application.  Thank you Mike and the others who have contributed to the creation and maintenance of the code.  It is great to have access to my own electricity consumption data in Home Assistant.  I hope this very useful HA integration is maintained well into the future. 

I implemented the integration on June 17, 2023 with a VUE2 and 16 CT sensors on my electrical panel.  The HA energy day graph for the 24 hours on June 18 and 19 presented the data accurately each hour over the 24 hours.

Unfortunately a problem occurred in the Energy day graph on June 20th and then again on June 21.  

On June 20, the problem is a wrong energy value of 14.78 kwh between 1200am and 100am.  Also between 100am and 200am there is a wrong energy value of 13.59 kwh returned to the grid.  The energy consumed values for the following periods in the day graph seem to be accurate. 

For June 21, the error is 12.91 kwh consumed between 1200am and 100am.  Also an error of a total returned 11.5 kwh between 100am and 200am.  The energy consumed values for the remaining hours for June 21 look OK.

My electrical panel is not set up to return any energy to the grid.  Normally my house is consuming about 0.60 - 0.50 kwh per hour between 1200am and 9:00am each day.

I notice that my total June 19 consumption was 14.78 kwh and it looks like this was carried over into June 20 for the period 1200am and 100am.

I believe I did a HA OS upgrade on June 19th.   I am new to HA and did not keep any notes on the exact date of the upgrade. Normally I track that information for troubleshooting.  The current versions of my setup are shown below. 

A PDF is attached that illustrates the graph for June 19 and June 20, 2023.

Home Assistant 2023.6.2
Supervisor 2023.06.2
Operating System 10.3
Frontend 20230608.0 - latest

[energy day graph problem June 20, 2023.pdf](https://github.com/magico13/ha-emporia-vue/files/11829145/energy.day.graph.problem.June.20.2023.pdf)

Ron

**Comment by Ron1926** [comment]
Hello 
For some reason the "energy return to grid" problem did not occur this morning on Jun 22, 2023.  I don't know how it was resolved.  I did not make any changes in any of the settings.  Hopefully the problem does not repeat.  Attached is a another PDF that illustrates the problem over a span of 4 days.
Ron
[energy return to grid problem - resolved on Jun 22, 2023.pdf](https://github.com/magico13/ha-emporia-vue/files/11836469/energy.return.to.grid.problem.-.resolved.on.Jun.22.2023.pdf)

**Comment by magico13** [comment]
This looks to be the same issue as #150 and is caused by the Emporia API reporting old data after midnight. It should balance out for the day but the data for the first hour or two in the energy graph can get messed up. It'll appear like in the picture below, with a drop down for the reset, a spike up for the API giving bad data, then a drop again when the API returns to giving proper data. If the spike up happens before the hour and the drop down happens after the hour, then the energy graph will look like what you've seen. If it happens within the hour it will handle it correctly. Sometimes it doesn't happen at all.

Until I am able to add code to filter out this behavior (and thus not trust the data from the API) then this will continue to happen.

![image](https://github.com/magico13/ha-emporia-vue/assets/5289685/98c98c5c-eecd-42d4-9f10-274ed518db63)

**Comment by Ron1926** [comment]
Hi Mike,
Thank you for the explanation.  Again, thank you for creating this application for the VUE2.
I am new to Home Assistant and still learning my way around the program.
After reading your email, I looked closer at my 02fridge circuit graph for June 21 and 22, 2023.  I now see odd data values that can occur in a 24 hour cycle.  I circled them in red in the attached PDF.
If I understand your explanation correctly, it sounds like your code simple grabs the data from the Emporia database and the odd data values reside in the Emporia database.
I will adapt to this. :)
Ron
[02fridge circuit for June 21 and 22, 2023.pdf](https://github.com/magico13/ha-emporia-vue/files/11840777/02fridge.circuit.for.June.21.and.22.2023.pdf)
